### PR TITLE
Added functionality to list subscribers in a topic. Made changes for …

### PR DESCRIPTION
**Pull request by lfportal**
#3

### CODE REVIEW

File: spec/bifrost/topic_spec.rb [hunk 1]
```diff
@@ -15,6 +15,11 @@
     expect { invalid_topic.save }.to raise_error(TypeError)
   end
 
+  it 'should not accept names with spaces in them' do
+    invalid_topic = Bifrost::Topic.new('topic name')
+    expect { invalid_topic.save }.to raise_error(URI::InvalidURIError)
```
Comment (shirren) [inline spec/bifrost/topic_spec.rb]: @lfportal I think we should wrap this. URI::InvalidURIError makes no sense in the context of the Bifrost

File: spec/bifrost/topic_spec.rb [hunk 2]
```diff
@@ -56,16 +61,10 @@
...
-      tp = Bifrost::Topic.new('test_donotdelete')
-      tp.save unless bus.topic_exists?(tp)
-    end
-
-    after(:all) do
+    before(:each) do
       bus = Bifrost::Bus.new
-      tp = Bifrost::Topic.new('test_donotdelete')
-      tp.delete if bus.topic_exists?(tp)
+      bus.topics.each(&:delete)
```
Comment (shirren) [inline spec/bifrost/topic_spec.rb]: @lfportal we should also leave the queues in a clean state after the spec

File: spec/bifrost/topic_spec.rb [hunk 3]
```diff
@@ -88,10 +87,24 @@
...
-    invalid_topic = Bifrost::Topic.new('topic name')
-    expect { invalid_topic.save }.to raise_error(URI::InvalidURIError)
+    context 'with no subscribers' do
+      it 'should return no subscribers' do
+        expect(new_topic.subscribers).to be_empty
+      end
+    end
+
+    context 'with subscribers' do
+      before(:each) do
```
Comment (shirren) [inline spec/bifrost/topic_spec.rb]: @lfportal at the parent level are we clearing all the topics at the end of the specs?

File: lib/bifrost/topic.rb [hunk 4]
```diff
@@ -63,8 +70,8 @@ def remove_subscriber(subscriber)
       end
     end
 
-    def ==(another_topic)
-      self.name == another_topic.name && self.class == another_topic.class
+    def ==(other)
```
Comment (shirren) [inline lib/bifrost/topic.rb]: @lfportal for future reference please do not make changes outside the scope of the issue raised

**Review by shirren — CHANGES_REQUESTED** [review]

File: spec/bifrost/topic_spec.rb (hunk 2 above)
Comment (lfportal) [inline spec/bifrost/topic_spec.rb]: @shirren This logic sounds like it should be global spec functionality. I will put this in the spec helper

File: spec/bifrost/topic_spec.rb (hunk 3 above)
Comment (lfportal) [inline spec/bifrost/topic_spec.rb]: @shirren I can put this into the spec helper in a `before(:suite)` block

File: lib/bifrost/topic.rb (hunk 4 above)
Comment (lfportal) [inline lib/bifrost/topic.rb]: @shirren Ok

**Pull request merged**